packages/hardhat-core/src/internal/cli/cli.ts

```diff
@@ -85,11 +85,7 @@ async function main() {
 
     showStackTraces = hardhatArguments.showStackTraces;
 
-    if (
-      hardhatArguments.config === undefined &&
-      !isCwdInsideProject() &&
-      process.stdout.isTTY === true
-    ) {
+    if (hardhatArguments.config === undefined && !isCwdInsideProject()) {
```

Comment:
Why is the `isTTY` check removed? The reason for it was to avoid things like piping hardhat to hang forever.

Reply:
Right, I meant to call this out before review, thanks for noticing.

The reason it's removed is that when hardhat is run to generate the sample project, during the scope of this test, it's being run via a node.js child process, and there is no TTY in that case, so it wouldn't fall into the project generation logic without this change.

Reply:
Maybe `process.stdout.isTTY || process.env.HARDHAT_CREATE_SAMPLE_PROJECT_WITH_DEFAULTS !== undefined`?

Reply:
Addressed in https://github.com/nomiclabs/hardhat/pull/1686/commits/2a8901ff84eb2c2894df3e60bf63b4752d167487